go/libraries/doltcore/schema/column.go

```diff
@@ -89,15 +90,24 @@ type Column struct {
 // NewColumn creates a Column instance with the default type info for the NomsKind
 func NewColumn(name string, tag uint64, kind types.NomsKind, partOfPK bool, constraints ...ColConstraint) Column {
 	typeInfo := typeinfo.FromKind(kind)
-	col, err := NewColumnWithTypeInfo(name, tag, typeInfo, partOfPK, "", false, "", constraints...)
+	col, err := NewColumnWithTypeInfo(name, tag, typeInfo, partOfPK, "", "", false, false, "", constraints...)
 	if err != nil {
 		panic(err)
 	}
 	return col
 }
 
 // NewColumnWithTypeInfo creates a Column instance with the given type info.
-func NewColumnWithTypeInfo(name string, tag uint64, typeInfo typeinfo.TypeInfo, partOfPK bool, defaultVal string, autoIncrement bool, comment string, constraints ...ColConstraint) (Column, error) {
+func NewColumnWithTypeInfo(
```

Comment:
Maybe the signature of this function stays the same and we get a new function like:

```
func NewVirtualColumnWithTypeInfo(name string, tag uint64, typeInfo typeinfo.TypeInfo, partOfPK bool, defaultVal generatedVal string, autoIncrement bool, comment string, constraints ...ColConstraint)
```

or something like that...

Reply:
I'm just using the struct directly where it's needed now